cli/azd/internal/version.go

```diff
@@ -27,6 +28,11 @@ type AzdVersionSpec struct {
 	Commit  string `json:"commit"`
 }
 
+// IsDevVersion returns true if the current version of azd is an unreleased, developer version.
```

Comment:
It'd be good to also include private CI builds here to get telemetry from them as well. I'm thinking that we would do a special tag for unreleased bits